mcstas/src/instrument.y

```diff
@@ -1267,7 +1267,29 @@ component: removable cpuonly split "COMPONENT" instname '=' instref
 	    }
 	  }
         }
-        if ($13->linenum)   comp->extend= $13;  /* EXTEND block*/
+        if ($13->linenum) {
+	  if (comp->extend->linenum>0) {
+	    fprintf(stderr, "\n-----------------------------------------------------------\n");
+	    fprintf(stderr, "WARNING (COMPONENT %s): Existing (COPY) EXTEND block:\n", comp->name);
+	    List_handle liter = list_iterate(comp->extend->lines);
+	    List_handle liter2 = list_iterate($13->lines);
+	    char *line, *line2;
+	    fprintf(stderr, "\n  EXTEND %%{\n");
+	    while((line = list_next(liter))) {
+	      fprintf(stderr, "  %s",line);
+	    }
+	    list_iterate_end(liter);
+	    fprintf(stderr, "  %%}\n");
+	    fprintf(stderr, "\nwill be been overwritten by:\n");
```

Comment:
I think a slightly different wording of the warning might make it clearer. Something like
```
	    fprintf(stderr, "WARNING: Existing (COPY) EXTEND block in COMPONENT %s:\n", comp->name);
	    List_handle liter = list_iterate(comp->extend->lines);
	    List_handle liter2 = list_iterate($13->lines);
	    char *line, *line2;
	    fprintf(stderr, "  EXTEND %%{\n");
	    while((line = list_next(liter))) {
	      fprintf(stderr, "  %s",line);
	    }
	    list_iterate_end(liter);
	    fprintf(stderr, "  %%}\n");
	    fprintf(stderr, "\nis overwritten by:\n");
```
(Plus I would either remove the extra line breaks, like I've done here, or remove the two-space indentation of the printed `EXTEND` blocks)
e.g.,
```cmd
WARNING: Existing (COPY) EXTEND block in COMPONENT named_instance:
  EXTEND %{
  ...
  %}
is overwritten by
  EXTEND %{
  ...
  %}
```

Reply:
Thanks @g5t, will implement that.